kubelet: Move common labels out of dockertools package

sjpotter opened the pull request:
moved labels and their Get functions to types

aliases kubernetes/types import to kubetypes to use kubelet/types as types

sjpotter commented:
@yifan-gu @euank @Random-Liu

Random-Liu commented:
@sjpotter PR needs rebase. :)

yifan-gu commented:
@sjpotter Can you clean up your github first comment? It will be copied into commits when it's merged.

dchen1107 commented:
Please rebase it.

Looks good to me overall. Thanks for cleaning up.

yifan-gu commented:
LGTM according to https://github.com/kubernetes/kubernetes/pull/25688#issuecomment-219545432
Feel free to hold this if there's any concern from your side @Random-Liu @vishh @timstclair

yifan-gu commented:
```
++ [0517 15:04:17] linux/amd64: go build started
# k8s.io/kubernetes/pkg/kubelet/server/stats
pkg/kubelet/server/stats/summary.go:222: undefined: "k8s.io/kubernetes/pkg/kubelet/types".GetContainerName
pkg/kubelet/server/stats/summary.go:247: undefined: "k8s.io/kubernetes/pkg/kubelet/types".GetPodName
pkg/kubelet/server/stats/summary.go:248: undefined: "k8s.io/kubernetes/pkg/kubelet/types".GetPodNamespace
pkg/kubelet/server/stats/summary.go:249: undefined: "k8s.io/kubernetes/pkg/kubelet/types".GetPodUID
pkg/kubelet/server/stats/summary.go:255: undefined: "k8s.io/kubernetes/pkg/kubelet/types".GetPodName
pkg/kubelet/server/stats/summary.go:256: undefined: "k8s.io/kubernetes/pkg/kubelet/types".GetPodNamespace
# k8s.io/kubernetes/pkg/kubelet/rkt
pkg/kubelet/rkt/rkt.go:575: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodUIDLabel
pkg/kubelet/rkt/rkt.go:576: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNameLabel
pkg/kubelet/rkt/rkt.go:577: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNamespaceLabel
pkg/kubelet/rkt/rkt.go:752: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesContainerNameLabel
pkg/kubelet/rkt/rkt.go:799: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodUIDLabel
pkg/kubelet/rkt/rkt.go:816: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodUIDLabel
pkg/kubelet/rkt/rkt.go:1196: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodUIDLabel
pkg/kubelet/rkt/rkt.go:1198: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodUIDLabel
pkg/kubelet/rkt/rkt.go:1200: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNameLabel
pkg/kubelet/rkt/rkt.go:1202: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNameLabel
pkg/kubelet/rkt/rkt.go:1202: too many errors
# k8s.io/kubernetes/pkg/kubelet/dockertools
pkg/kubelet/dockertools/labels.go:69: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNameLabel
pkg/kubelet/dockertools/labels.go:70: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNamespaceLabel
pkg/kubelet/dockertools/labels.go:71: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodUIDLabel
pkg/kubelet/dockertools/labels.go:79: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesContainerNameLabel
pkg/kubelet/dockertools/labels.go:106: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNameLabel
pkg/kubelet/dockertools/labels.go:107: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodNamespaceLabel
pkg/kubelet/dockertools/labels.go:108: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesPodUIDLabel
pkg/kubelet/dockertools/labels.go:109: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesContainerNameLabel
pkg/kubelet/dockertools/manager.go:1365: undefined: "k8s.io/kubernetes/pkg/kubelet/types".KubernetesContainerNameLabel
!!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:435
  'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${goldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:435 kube::golang::build_binaries_for_platform(...)
  2: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:555 kube::golang::build_binaries(...)
  3: hack/build-cross.sh:28 main(...)
Exiting with status 1
```

Random-Liu commented:
@sjpotter I only saw you remove the original labels, but where did you move them? Looks like some changes are missing.

sjpotter commented:
I moved them to kubelet/types and it previous built, unsure what happened
here, will take a look at it later

On Tue, May 17, 2016 at 11:45 PM, Lantao Liu <notifications@github.com>
wrote:

> @sjpotter <https://github.com/sjpotter> I only saw you remove the
> original labels, but where did you move them? Looks like some changes are
> missing.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25688#issuecomment-219939306>
>

sjpotter commented:
rebaed to master and still make quick-release works fine. going to do a make release, but dont see what that should make a difference for this (especially as the above error is linux/amd64)

sjpotter commented:
figured it out, :(

sjpotter commented:
these tests that are failing seem to have nothing to do with changes in this PR, are they flakey?

Random-Liu commented:
```
FAIL	k8s.io/kubernetes/pkg/kubelet/rkt [build failed]
```

sjpotter commented:
have to update docker and rkt tests that use the old variable names

euank commented:
This PR is going to conflict with #25326 ... and in fact those labels are not really going to be used anymore it looks like. Is this PR still useful in that light?

euank commented on pkg/kubelet/dockertools/labels.go:
```diff
@@ -24,23 +24,20 @@ import (
 
-	kubernetesContainerNameLabel                   = "io.kubernetes.container.name"
 	kubernetesContainerHashLabel                   = "io.kubernetes.container.hash"
 	kubernetesContainerRestartCountLabel           = "io.kubernetes.container.restartCount"
```
Note these other labels (e.g. restartcount, terminationMessagePath) are also in rkt, it just happens that they had to be duplicated before due to casing issues. Maybe all of the labels should be moved, not just a few?

sjpotter replied:
I only wanted to move labels that are used outside of the container runtimes for now.  while these labels are used in both, they are not things consumed in the kubelet itself.

sjpotter replied:
also, can't move some due to camel case

yifan-gu replied:
@euank, so @Random-Liu has a pending PR to partially fix this (changing labels to lower-cases, but not moving them out into a common package) https://github.com/kubernetes/kubernetes/pull/25326

yifan-gu replied:
Which means these two PRs will rebase on each other once the other one gets merged cc @Random-Liu

sjpotter commented:
at minimum we need to change the annotations that rktnetes uses to the same as docker for these 4 annotations (including adding container name)

sjpotter commented:
is this ready to go?  @Random-Liu

Random-Liu commented on pkg/kubelet/dockertools/manager.go:
```diff
@@ -48,12 +48,12 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/network/hairpin"
 	proberesults "k8s.io/kubernetes/pkg/kubelet/prober/results"
 	"k8s.io/kubernetes/pkg/kubelet/qos"
-	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
```
We call pkg/kubelet/types as kubetypes before, why do we change it?

dchen1107 replied:
In my opinion, I am ok with this change since with Shaya's new change, kubetype is denoted for the common types for entire kubernetes while pkg/kubelet/types only includes kubelet internal types.

sjpotter replied:
thats why I did it, I can submit again with it reverted, but about to go to dentist, so just tell me what you guys want.

Random-Liu replied:
@sjpotter @dchen1107 If so, we should also change `kubecontainer` and `kubepod` in the future. :)

sjpotter replied:
Ill do this janitorial work later today
On May 19, 2016 2:27 PM, "Lantao Liu" <notifications@github.com> wrote:

> In pkg/kubelet/dockertools/manager.go
> <https://github.com/kubernetes/kubernetes/pull/25688#discussion_r63954059>
> :
>
> > @@ -48,12 +48,12 @@ import (
> >  	"k8s.io/kubernetes/pkg/kubelet/network/hairpin"
> >  	proberesults "k8s.io/kubernetes/pkg/kubelet/prober/results"
> >  	"k8s.io/kubernetes/pkg/kubelet/qos"
> > -	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
>
> @sjpotter <https://github.com/sjpotter> @dchen1107
> <https://github.com/dchen1107> If so, we may also want to change
> kubecontainer and kubepod in the future. :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25688/files/d2700058430e71b4046bc1886fd99e808bee090f#r63954059>
>

Random-Liu commented:
@sjpotter @yifan-gu we should name "pkg/kubelet/types" to `kubetypes` instead of "pkg/api/types". Other than this, LGTM.

sjpotter commented:
ok, I thought they should be the opposite, but ok, I'll flip them.

Random-Liu commented:
LGTM, and we should change the name of other packages in the future https://github.com/kubernetes/kubernetes/pull/25688#discussion_r63954059

sjpotter commented:
so it turns out that the janitorial work is a bit harder than it seems.

for instance "kubelet/pod" import is aliased as kubepod, this is all well and good, and is neccessary because many of the files that import it have a variable that is idiomatically called pod.

need to make a separate issue talking about this

dchen1107 commented:
@sjpotter Please file a separate issue for that. LGTM this one.

sjpotter commented:
done https://github.com/kubernetes/kubernetes/issues/25932

Random-Liu commented:
@sjpotter Thanks a lot! ;)

sjpotter committed:
move common lables used outside of containr runtime out of dockertools

moved labels and their Get functions to types

aliases kubernetes/types import to kubetypes to use kubelet/types as types

sjpotter commented:
@yifan-gu had to change to handle some of rkt.go changes, but all just import name changes, can you add label back?

sjpotter commented:
@dchen1107 need a LGTM reapplied to this a in rebase hell with submit queue.

dchen1107 commented:
Looks like jenkins e2e failed for you, but not due to your changes. :-( 

@k8s-bot ok to test

dchen1107 commented:
LGTM

sjpotter commented:
@dchen1107 looks like same failure. unsure whats going wrong here

sjpotter commented:
@k8s-bot test this

sjpotter commented:
@dchen1107 it lost its LGTM, its now passing tests again w/o any changes

Random-Liu commented:
LGTM.

sjpotter commented:
thanks

sjpotter commented:
as far as I can tell, the unit test that fails passes for me.

from unit/integration log

FAIL	k8s.io/kubernetes/pkg/controller/volume/reconciler	1.532s

from my test on a rebased (nothing  changed) tree

$ go test ./pkg/controller/volume/reconciler
ok  	k8s.io/kubernetes/pkg/controller/volume/reconciler	0.097s

@dchen1107 @Random-Liu

yujuhong commented:
@k8s-bot test this issue #26377

yujuhong commented:
@k8s-bot unit test this issue #26377
